wordpress-mobile/WordPress-Android
# Site Creation: Site Info VM, Fragment and Unit Tests

Fixes #8503, #8504, #8535. This PR adds the fragment, the view model and the unit tests for the new site info screen. The screen itself is pretty basic and the only communication necessary from fragment to vm is the update to `businessName` and the `tagLine` and the only communication back is the state of the `skip/next` button.

The PR follows the same patterns used by the verticals and segments screens created by @malinajirka. It does not connect to the main VM yet which will be done in a separate PR depending on how we want to handle the loading.

Here are the screenshots for the screen in empty and filled states in both portrait and landscape orientations for both a phone and a tablet: https://cloudup.com/c-jAAw9KjmW @SylvesterWilmott If I missed any screenshots, let me know and I can add it for you.

To test:
* The `NEW_SITE_CREATION_ENABLED` flag needs to be enabled in `WordPress/build.gradle`.
* Since the fragment is not added to the flow yet, it'll need to be done to be able to test. One easy way to do that would be to change the `showFragment` in `NewSiteCreationActivity.onCreate()` to instead show the newly added fragment. Once that's done, you can go to the site selection screen, tap on the `+` icon and select "Create WordPress.com site" to see it.
* Make sure that the `Skip` button is visible initially and whenever both the business name and the tag line is empty
* Make sure the `Next` button is visible whenever either the business name or the tag line is filled
* Make sure the position of the `Next/Skip` button is on the bottom right of the page and pushed during keyboard transitions. When the button is not visible in the screen, make sure you can scroll to it and the position makes sense.
* Remember to rotate your screen once in a while.
